openfasoc/generators/common/simulation/classify_sim_error.py

```diff
@@ -0,0 +1,55 @@
+import warnings
+import sys
+import os
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from common.get_ngspice_version import check_ngspice_version
+            
+def classify_sim_error(template_file, result_file, _generator_is) -> str:
+    """Used to propogate out how close the generated simulations results are 
+    from the results in the stored template files
+
+    Args:
+    - 'template_file' (string): The stored template file
+    - 'result_file' (string): The result file generated by the simulations
+    - '_generator_is' (dict): Contains info about the generator for which the simulation check is being run
+
+    Returns:
+    - str: The kind of alert
+        - 'red' alert for very large difference in generated and template results
+        - 'amber' alert if the difference lies within the allowable range
+        - 'green' if ok
+    """
+    if _generator_is['sky130hd_temp']:
+        sys.path.append(os.path.join(os.path.dirname(__file__), '../temp-sense-gen'))
+        
+    elif _generator_is['sky130hvl_ldo']:
```

Comment:
Is it possible to do this in generator specific files instead? I believe the `compare_files` function is imported from the generator right? I think it would be better to take this function as an argument for `classify_sim_error` instead and import the files in the workflow. This way the common API will have no generator specific code.

Reply:
From the most recent commit on, the `compare_files` function will be passed and the module import will be done from `.github/scripts/parse_rpt.py` itself, to avoid exposing the `generator_is` variable